src/ra_server.erl

```diff
@@ -192,6 +194,18 @@
 %% The simple machine config is version that can only be used for simple state
 %% machines that cannot access any of the advanced features.
 
+-type upgrade_machine_when() :: after_election | when_available_everywhere.
```

Comment:
we could consider `-type machine_upgrade_strategy() :: all | quorum.`

Reply:
Suggestion applied :)